CRM/Upgrade/Incremental/php/FourFive.php

```diff
@@ -303,6 +303,57 @@ static function migrateHonoreeInfo(CRM_Queue_TaskContext $ctx) {
   }
 
   /**
+   * Upgrade function.
+   *
+   * @param string $rev
+   */
+  public function upgrade_4_5_6($rev) {
+    // Task to process sql.
+    $this->addTask(ts('Upgrade DB to 4.5.6: Fix saved searches consisting of multi-choice custom field(s)'), 'updateSavedSearch');
+
+    return TRUE;
+  }
+
+  /**
+   * Update saved search for multi-select custom fields on DB upgrade
+   *
+   * @param CRM_Queue_TaskContext $ctx
+   *
+   * @return bool TRUE for success
+   */
+  static function updateSavedSearch(CRM_Queue_TaskContext $ctx) {
+    $sql = "SELECT id, form_values FROM civicrm_saved_search";
+    $dao = CRM_Core_DAO::executeQuery($sql);
+    while ($dao->fetch()) {
+      $formValues = unserialize($dao->form_values);
+      while (list($field, $data_value) = each($formValues)) {
```

Comment:
What's the reason for not using `foreach` here? Seems like `list` will throw an error at the end of the loop because you can't `list FALSE`.